recipes/vtk/all/conanfile.py

```diff
@@ -0,0 +1,445 @@
+
+        # Should we add below defines depending on "group_qt"? Or rather that should be driven by package consumer "vtk:module_guisupportqt=True" and "vtk:module_renderingqt=True"?
+        self._cmake.definitions["VTK_MODULE_ENABLE_VTK_GUISupportQt"] = "YES" if self.options.group_qt else "DEFAULT"
+        self._cmake.definitions["VTK_MODULE_ENABLE_VTK_RenderingQt"] = "YES" if self.options.group_qt else "DEFAULT"
+        # Looks like "VTK_MODULE_ENABLE_VTK_GUISupportQtOpenGL" doesn't exist in VTK 9.0.1
+        # self._cmake.definitions["VTK_MODULE_ENABLE_VTK_GUISupportQtOpenGL"] = "YES" if self.options.group_qt else "NO"
+
+        # I believe introducing _groups and _modules, makes this code deprecated.
+        #       Leave it temporarily until tests are finished
+        # self._cmake.definitions["VTK_Group_StandAlone"] = "OFF" if self.options.minimal else "ON"
+        # self._cmake.definitions["VTK_Group_Rendering"] = "OFF" if self.options.minimal else "ON"
+        #
+        # self._cmake.definitions["Module_vtkIOXML"] = "ON" if self.options.ioxml else "OFF"
+        #
+        # self._cmake.definitions["Module_vtkIOExport"] = "ON" if self.options.ioexport else "OFF"
+        #
+        # self._cmake.definitions["VTK_Group_Qt"] = "ON" if self.options.with_qt else "OFF"
+        # self._cmake.definitions["VTK_MODULE_ENABLE_VTK_GUISupportQt"] = "YES" if self.options.with_qt else "NO"
+        # self._cmake.definitions["VTK_MODULE_ENABLE_VTK_GUISupportQtOpenGL"] = "YES" if self.options.with_qt else "NO"
+        # self._cmake.definitions["VTK_MODULE_ENABLE_VTK_RenderingQt"] = "YES" if self.options.with_qt else "NO"
+        #
+        # self._cmake.definitions["VTK_Group_MPI"] = "ON" if self.options.mpi else "OFF"
+        # self._cmake.definitions["Module_vtkIOParallelXML"] = "ON" if self.options.mpi else "OFF"
+        #
+        # self._cmake.definitions["Module_vtkIOParallelXML"] = "ON" if self.options.mpi_minimal else "OFF"
+        # self._cmake.definitions["Module_vtkParallelMPI"] = "ON" if self.options.mpi_minimal else "OFF"
+
+        # if self.settings.os == 'Macos':
+        #     self.env['DYLD_LIBRARY_PATH'] = os.path.join(self.build_folder, 'lib')
+        #     self.output.info("cmake build: %s" % self.build_folder)
+
+        self._cmake.configure(build_folder='build')
+        return self._cmake
+
+    def build(self):
+        # if self.options.with_qt:
+        if self.options.group_qt:
+            if self.options["qt"].shared == False:
+                raise ConanInvalidConfiguration("VTK option 'group_qt' requires 'qt:shared=True'")
+            if self.settings.os == "Linux":
+                if self.options["qt"].qtx11extras == False:
+                    raise ConanInvalidConfiguration("VTK option 'group_qt' requires 'qt:qtx11extras=True'")
+                
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
```

Comment:
@madebr, @Croydon
Conan experts!

Is it better to have patching here, in `build()` method or it rather should be in the `source()` method?

Reply:
Patching should happen ind `build()`

Reply:
Thank You!
Done.